lua/autorun/client/b-draw_lib.lua

```diff
@@ -55,48 +57,43 @@ local function fetch_asset(url, fallback)
 	return mats[url]
 end
 
-local function fetchAvatarAsset(id64, size, onFetched)
-	id64 = id64 or "BOT"
-	size = size == "medium" and "medium" or size == "small" and "" or size == "large" and "full" or ""
-
-	if fetchedavatars[id64 .. " " .. size] then
-		return fetchedavatars[id64 .. " " .. size]
+local function FetchAvatarAsset(id64, size, fallback)
+	if not id64 then
+		return _bot_avatar
 	end
 
-	fetchedavatars[id64 .. " " .. size] = id64 == "BOT" and "http://steamcdn-a.akamaihd.net/steamcommunity/public/images/avatars/09/09962d76e5bd5b91a94ee76b07518ac6e240057a_full.jpg" or "http://i.imgur.com/uaYpdq7.png"
+	size = size == "medium" and "_medium" or size == "large" and "_full" or ""
+	fallback = fallback or _default_avatar
 
-	if id64 == "BOT" then
-		if isfunction(onFetched) then
-			onFetched(fetchedavatars[id64 .. " " .. size])
-		end
+	local key = id64 .. size
 
-		return
+	if fetched_avatar_urls[key] then
+		return FetchAsset(fetched_avatar_urls[key], fallback)
 	end
 
 	fetch("http://steamcommunity.com/profiles/" .. id64 .. "/?xml=1", function(body)
-		local link = body:match("https://steamcdn%-a%.akamaihd%.net/steamcommunity/public/images/avatars/.-%.jpg") -- fix this with new https and gmod regex
-		if not link then return end
+		local link = body:match("<avatarIcon><%!%[CDATA%[(.*)%]%]><%/avatarIcon>")
 
-		fetchedavatars[id64 .. " " .. size] = link:Replace(".jpg", (size ~= "" and "_" .. size or "") .. ".jpg")
+		if not link then return end
 
-		if isfunction(onFetched) then
-			onFetched(fetchedavatars[id64 .. " " .. size])
-		end
+		fetched_avatar_urls[key] = link:Replace(".jpg", size .. ".jpg")
+		FetchAsset(fetched_avatar_urls[key])
```

Comment:
Why is this fetched here? (if legitimate, add a comment explaining it)

Reply:
I'm fetching them here because they wouldn't be cached elsewise.
Since fetching is asynchronous i want to have them fetched before they are first used.

(I copied this behaviour from the old function)